src/main/java/tdl/client/queue/QueueBasedImplementationRunner.java

```diff
@@ -1,29 +1,214 @@
+
+    private static final String ATTRIBUTE_EVENT_NAME = "name";
+    private static final String ATTRIBUTE_EVENT_VERSION = "version";
+
+    private final ObjectMapper mapper;
+
+    private AmazonSQS client;
+    private String serviceEndpoint;
+
+    private ReceiveMessageRequest receiveRequestQueueMessage;
+
+    private CreateQueueRequest messageRequestQueue;
+    private String messageRequestQueueUrl;
+    private CreateQueueRequest messageResponse;
+    private String messageResponseQueueUrl;
+    private final Gson gson;
+
+    private DeleteMessageBatchRequest batchOfMessagesToDeleteRequest;
+    private List<DeleteMessageBatchRequestEntry> deleteMessageBatchRequestEntries;
+
     private QueueBasedImplementationRunner(ImplementationRunnerConfig config, ProcessingRules deployProcessingRules) {
+        logToConsole("        QueueBasedImplementationRunner creation [start]");
         this.config = config;
         this.deployProcessingRules = deployProcessingRules;
         audit = new Audit(config.getAuditStream());
+
+        requests = new ArrayList<>();
+
+        logToConsole("     QueueBasedImplementationRunner creation [start]");
+
+        String hostname = config.getHostname();
+        int port = config.getPort();
+
+        Config customConfig = ConfigFactory.load();
+        customConfig.checkValid(ConfigFactory.defaultReference());
+        serviceEndpoint = String.format("http://%s:%d", hostname, port);
+        client = createAWSClient(
+                serviceEndpoint,
+                customConfig.getString("sqs.signingRegion"),
+                customConfig.getString("sqs.accessKey"),
+                customConfig.getString("sqs.secretKey")
+        );
+
+        try {
+            messageRequestQueue = createQueueRequest(customConfig.getString("sqs.requestQueueName"));
+            messageRequestQueueUrl = createQueue(messageRequestQueue);
+            messageResponse = createQueueRequest(customConfig.getString("sqs.responseQueueName"));
+            messageResponseQueueUrl = createQueue(messageResponse);
+        } catch (Exception ex) {
+            logToConsole("        QueueBasedImplementationRunner [error]");
+            String message = "There was a problem creating the queue runner";
+            LOGGER.error(message, ex);
+            audit.logException(message, ex);
+        }
+
+        receiveRequestQueueMessage = createReceiveRequest(messageRequestQueueUrl);
+
+        mapper = new ObjectMapper();
+
+        GsonBuilder gsonBuilder = new GsonBuilder();
+        gsonBuilder.registerTypeAdapter(Double.class,
+                (JsonSerializer<Double>) (src, typeOfSrc, context) -> {
+                    Integer value = (int) Math.round(src);
+                    return new JsonPrimitive(value);
+                });
+
+        gson = gsonBuilder.serializeNulls().create();
+
+        batchOfMessagesToDeleteRequest = new DeleteMessageBatchRequest().withQueueUrl(messageResponseQueueUrl);
```

Comment:
Is this the right queue? `messageResponseQueueUrl` I think we need to delete messages off the `messageRequestQueueUrl`